### ISSUE TITLE
kubectl exec fails because of missing log directory

### ISSUE THREAD

**Issue by pcallewaert** [gravitational/teleport#5014]
### Description

**What happened**:

I am using the new v5 kubernetes functionality, and the `teleport-kube-agent` helm chart from the examples directory.
Unfortunately, when I used `kubectl exec` no shell was opened.

I see the following in the logs of the main teleport server:
```txt
Dec  1 08:09:21 we-jumpstation-teleport-vm0 teleport[17934]: INFO [AUDIT]     kube.request addr.local:10.0.0.1:443 addr.remote:<CENSURED>:3024 code:T3009I ei:0 event:kube.request kubernetes_cluster:aks-ndt-internal-westeurope-production kubernetes_groups:[engineering system:authenticated system:masters] kubernetes_users:[pcallewaert] login:pcallewaert namespace:default proto:kube request_path:/api/v1/namespaces/dev/pods/dev-influxdb-d95776749-rjpv8 resource_api_group:core/v1 resource_kind:pods resource_name:dev-influxdb-d95776749-rjpv8 resource_namespace:dev response_code:200 server_id:28b4baed-1524-4ce1-93be-0e4dca6d495c time:2020-12-01T08:09:21.615Z uid:fe120b15-4bde-4d95-827f-79834b2f32ba user:pcallewaert verb:GET events/emitter.go:318
Dec  1 08:09:21 we-jumpstation-teleport-vm0 teleport[17934]: INFO             Negotiated protocol v4.channel.k8s.io. proxy/remotecommand.go:127
Dec  1 08:09:21 we-jumpstation-teleport-vm0 teleport[17934]: WARN [PROXY:PRO] Executor failed while streaming. error:path "/var/lib/teleport/log/upload/streaming/default" does not exist or is not a directory proxy/forwarder.go:773
Dec  1 08:09:21 we-jumpstation-teleport-vm0 teleport[17934]: 2020/12/01 08:09:21 http: response.WriteHeader on hijacked connection from github.com/gravitational/trace.replyJSON (httplib.go:87)
Dec  1 08:09:21 we-jumpstation-teleport-vm0 teleport[17934]: 2020/12/01 08:09:21 http: response.Write on hijacked connection from github.com/gravitational/trace.replyJSON (httplib.go:108)
```

**What you expected to happen**:
A shell to the pod should be opened

**How to reproduce it (as minimally and precisely as possible)**:
Using the teleport-kube-agent chart from the examples directory.

**Workaround**:
I did get it working by creating manually the following directory in the kube-agent:
```sh
mkdir -p /var/lib/teleport/log/upload/streaming/default
```
I'm thinking as temporary fix to patch the kubernetes deployment so an emptyDir volume is mounted to that directory so its created.

### Environment
- Teleport version (use `teleport version`): v5.0.0
- Tsh version (use `tsh version`): v5.0.0
- OS (e.g. from `/etc/os-release`): helm-chart

- Where are you running Teleport? (e.g. AWS, GCP, Dedicated Hardware):
AKS

**Relevant Debug Logs If Applicable**

I updated from v4.3.5, and was working with leaf clusters. First upgraded to v5.0, added kube-agent to the kubernetes clusters, and removed the leaf clusters afterwards.

- `tsh --debug`
- `teleport --debug`

**Comment by pcallewaert** [comment]
The workaround is not great, at least we can `exec` to the pods, but the session recordings are not available in the WebUI.

I see the following error in the teleport-kube-agent:
```txt
E1201 08:44:36.978565       6 v2.go:105] EOF
E1201 08:46:57.038307       6 v2.go:105] EOF
ERRO             Failed to emit audit event session.end(T2004I). error:context canceled or closed events/emitter.go:468
```

**Comment by awly** [comment]
@pcallewaert can you share a redacted `teleport.yaml` for the main teleport server?

**Comment by pcallewaert** [comment]
```yaml
teleport:
  nodename: we-jumpstation-teleport-vm0.my.teleport.domain
  data_dir: /var/lib/teleport
  pid_file: /var/run/teleport.pid
  auth_token: cluster-join-token
  connection_limits:
    max_connections: 15000
    max_users: 250
  log:
    output: stderr
    severity: DEBUG
  ca_pin: ""
  advertise_ip: my-ip
auth_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3025
  client_idle_timeout: 0s
  disconnect_expired_cert: false
  keep_alive_count_max: 0
  cluster_name: "nephroflow"
  authentication:
    type: github
ssh_service:
  enabled: "no"
proxy_service:
  enabled: "yes"
  listen_addr: 0.0.0.0:3023
  web_listen_addr: 0.0.0.0:3080
  tunnel_listen_addr: 0.0.0.0:3024
  public_addr: my.teleport.domain:3080
  ssh_public_addr: my.teleport.domain:3023
  https_key_file: /etc/letsencrypt/live/my.teleport.domain/privkey.pem
  https_cert_file: /etc/letsencrypt/live/my.teleport.domain/fullchain.pem
  kubernetes:
    enabled: yes
    public_addr: ["my.teleport.domain:3026"]
```

Also, thanks for the heads-up about upgrading sequence

**Comment by awly** [comment]
@pcallewaert thanks. I was able to reproduce the issue.
Investigating why the recording folder auto-create fails...

Also, the error about emitted event is another bug, trivial to fix.
I'll try to get a batch of fixes in this week, and cut 5.0.1.

**Comment by awly** [comment]
This has landed in 5.0.1.
Please upgrade and reopen if you still experience this issue
